apps/remote_control/lib/lexical/remote_control/search/store/mnesia/schema.ex

```diff
@@ -0,0 +1,171 @@
+      :ok ->
+        State.rpc_call(state, :mnesia, :change_config, [:extra_db_nodes, [Node.self()]])
+
+      {:error, {node, {:already_exists, node}}} ->
+        :ok
+    end
+  end
+
+  def create_table(%State{} = state) do
+    case State.rpc_call(state, :mnesia, :create_table, [Mnesia, table_options(state)]) do
+      {:atomic, :ok} ->
+        :ok
+
+      {:aborted, {:already_exists, Mnesia}} ->
+        :ok
+
+      error ->
+        error
+    end
+  end
+
+  def to_key(subject, type, subtype, elixir_version \\ nil, erlang_version \\ nil) do
+    versions = Versions.current()
+    elixir_version = elixir_version || versions.elixir
+    erlang_version = erlang_version || versions.erlang
+    {to_subject(subject), type, subtype, elixir_version, erlang_version}
+  end
+
+  def to_row(%Entry{} = entry) do
+    entry_key =
+      to_key(entry.subject, entry.type, entry.subtype, entry.elixir_version, entry.erlang_version)
+
+    row(key: entry_key, path: entry.path, reference: entry.ref, value: entry)
+  end
+
+  # Private
+  defp to_subject(binary) when is_binary(binary), do: binary
+  defp to_subject(:_), do: :_
+  defp to_subject(other), do: inspect(other)
+
+  defp create_schema(%State{} = state) do
+    mnesia_node = state.mnesia_node
+
+    case State.rpc_call(state, :mnesia, :create_schema, [[state.mnesia_node]]) do
+      :ok ->
+        :ok
+
+      {:error, {^mnesia_node, {:already_exists, ^mnesia_node}}} ->
+        :ok
+
+      other_error ->
+        other_error
+    end
+  end
+
+  defp create_local_ram_table_copies(%State{} = state) do
+    if Mnesia.persist_to_disc?() and Node.self() not in existing_ram_table_copies(state) do
+      do_create_local_ram_table_copies(state)
+    else
+      :ok
+    end
+  end
+
+  defp do_create_local_ram_table_copies(%State{} = state) do
+    case State.rpc_call(state, :mnesia, :add_table_copy, [Mnesia, Node.self(), :ram_copies]) do
+      {:atomic, :ok} ->
+        :ok
+
+      {:aborted, {:already_exists, _, _}} ->
+        :ok
+
+      error ->
+        error
+    end
+  end
+
+  defp existing_ram_table_copies(%State{} = state) do
+    case State.rpc_call(state, :mnesia, :table_info, [Mnesia, :ram_copies]) do
+      {:badrpc, {:EXIT, {:aborted, {:no_exists, Mnesia, :ram_copies}}}} ->
+        []
+
+      list when is_list(list) ->
+        list
+    end
+  end
+
+  defp table_options(%State{} = state) do
+    defaults = [
+      attributes: [:key | @fields],
+      index: [:path, :reference],
+      type: :bag
+    ]
+
+    if Mnesia.persist_to_disc?() do
+      defaults
+      |> Keyword.put(:disc_copies, [state.mnesia_node])
+      |> Keyword.put(:ram_copies, [Node.self()])
+    else
+      defaults
+      |> Keyword.put(:disc_copies, [])
+      |> Keyword.put(:ram_copies, [state.mnesia_node, Node.self()])
+    end
+  end
+
+  defp mnesia_dir(%Project{} = project) do
+    project
+    |> Project.workspace_path(Path.join(["indexes", "mnesia"]))
```

Comment:
```suggestion
    |> Project.workspace_path("indexes/mnesia")
```

Reply:
I use `Path.join` to be cross-platform.

Reply:
Ah, good point! Maybe `Project.workspace_path` could accept a list as its second arg, but that could be discussed/changed later.